kube-client/src/config/mod.rs

```diff
@@ -155,6 +155,9 @@ pub struct Config {
     // TODO Actually support proxy or create an example with custom client
     /// Optional proxy URL.
     pub proxy_url: Option<http::Uri>,
+    /// If set, apiserver certificate will be validated to contain this string
+    /// (instead of hostname or IP address from the url).
```

Comment:
So the `Config` option exposed here is now technically only used to signal whether to opt-in to extra rustls behaviour. If i understand correctly it might be worth either:

- pass the `tls_server_name` into whatever is needed on the openssl side
- raise an issue to support tls_server_name on openssl (and reference it here)

Do you think that makes sense? I can imagine this property being useful on both tls stacks (outside of the rustls hack), but afaikt it also isn't really needed except to grant more security/strictness in the certificate validation stage.

Reply:
i'll raise an issue about supporting `tls_server_name` after merge so we remember.